x-pack/metricbeat/module/awsfargate/task_stats/task_stats.go

```diff
@@ -0,0 +1,202 @@
+)
+
+var metricsetName = "task_stats"
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each host defined in the module's configuration. After the
+// MetricSet has been created then Fetch will begin to be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet(awsfargate.ModuleName, metricsetName, New,
+		mb.DefaultMetricSet(),
+	)
+}
+
+// MetricSet holds any configuration or state information. It must implement
+// the mb.MetricSet interface. And this is best achieved by embedding
+// mb.BaseMetricSet because it implements all of the required mb.MetricSet
+// interface methods except for Fetch.
+type MetricSet struct {
+	*awsfargate.MetricSet
+	logger *logp.Logger
+}
+
+// New creates a new instance of the MetricSet. New is responsible for unpacking
+// any MetricSet specific configuration options if there are any.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	logger := logp.NewLogger(metricsetName)
+	metricSet, err := awsfargate.NewMetricSet(base)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating aws metricset")
+	}
+
+	return &MetricSet{
+		MetricSet: metricSet,
+		logger:    logger,
+	}, nil
+}
+
+// Fetch methods implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(report mb.ReporterV2) error {
+	// Get response from ${ECS_CONTAINER_METADATA_URI_V4}/task/stats
+	ecsURI, ok := os.LookupEnv("ECS_CONTAINER_METADATA_URI_V4")
+	if !ok {
+		err := fmt.Errorf("lookup $ECS_CONTAINER_METADATA_URI_V4 failed")
+		m.logger.Error(err)
+		return err
+	}
+
+	resp, err := http.Get(fmt.Sprintf("%s/task/stats", ecsURI))
+	if err != nil {
+		err := fmt.Errorf("http.Get failed: %w", err)
+		m.logger.Error(err)
+		return err
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		err := fmt.Errorf("ioutil.ReadAll failed: %w", err)
+		m.logger.Error(err)
+		return err
+	}
+
+	var output map[string]interface{}
+	err = json.Unmarshal(body, &output)
+	if err != nil {
+		err := fmt.Errorf("json.Unmarshal failed: %w", err)
+		m.logger.Error(err)
+		return err
+	}
+
+	var events []mb.Event
+	for queryID, taskStats := range output {
+		event := mb.Event{
+			ID:              queryID,
+			MetricSetFields: common.MapStr{},
+		}
+
+		taskMeta := taskStats.(map[string]interface{})
+		readTimestamp := taskMeta["read"].(string)
+		timestamp, err := time.Parse(time.RFC3339, readTimestamp)
+		if err != nil {
+			m.logger.Warn(fmt.Errorf("parsing timestamp %s failed, use current timestamp in event instead", readTimestamp))
+		} else {
+			event.Timestamp = timestamp
+		}
+
+		cpuStats := taskMeta["cpu_stats"].(map[string]interface{})
```

Comment:
Not sure if applicable but maybe we can use a `Schema` here like waht we do for https://github.com/elastic/beats/blob/95626b8f1690344312c0831ab2bdcbccffe4d089/metricbeat/module/nats/stats/data.go#L42

Reply:
Yep! I just added all schemas into data.go file.